spring-boot-kotlin-io/src/main/kotlin/com/baeldung/nonblockingcoroutines/config/DatastoreConfig.kt

```diff
@@ -0,0 +1,22 @@
+package com.baeldung.nonblockingcoroutines.config
+
+import io.r2dbc.postgresql.PostgresqlConnectionConfiguration
+import io.r2dbc.postgresql.PostgresqlConnectionFactory
+import io.r2dbc.spi.ConnectionFactory
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.data.r2dbc.config.AbstractR2dbcConfiguration
+import org.springframework.data.r2dbc.repository.config.EnableR2dbcRepositories
+
+
+@Configuration
+@EnableR2dbcRepositories
+class DatabaseConfig : AbstractR2dbcConfiguration() {
+
+    @Bean
+    override fun connectionFactory(): ConnectionFactory {
+        return PostgresqlConnectionFactory(PostgresqlConnectionConfiguration.builder()
+                // Set options here
```

Comment:
Does this example have any options to set? Do we need this comment?

Reply:
We need to provide database configuration options here in order to access a postgresql, in that case, instance. Locally, I used credentials for accessing my instance of PostgreSQL. 
As I was told by other authors, code examples should be compilable, but not always usable. 
What is the right way?